src/Extension/Challenge.cs

```diff
@@ -17,7 +17,7 @@ public class Challenge
         public Func<ChallengeContext, Task> OnCodeSubmittedHandler { get; private set; }
         public Evaluation CurrentEvaluation { get; private set; }
         public ChallengeSubmission CurrentSubmission => _submissionHistory.Peek();
-        public IEnumerable<ChallengeSubmission> SubmissionHistory => _submissionHistory;
+        public Stack<ChallengeSubmission> SubmissionHistory => _submissionHistory;
```

Comment:
I would not expose the stack here, the history should be immutable, that is why we did IEnumerable.

Reply:
My concern is that there is a hidden invariant in which the first item of the enumerable is the top of the stack. Is there a way to communicate the idea that a stack is the underlying data structure used?
The thinking that the last item in the enumerable is the last submission is also valid reasoning.

Reply:
the foreach on a stack is always consistent, exposing the stack makes it mutable. you can achieve all you want in exposing history as _history.Reverse()

Reply:
What if I use ImmutableStack history  => new ImmutableStack<>(_history)?